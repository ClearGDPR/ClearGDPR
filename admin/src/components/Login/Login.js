import React from 'react';
import PropTypes from 'prop-types';
import { Form, Text } from 'react-form';

import logo from 'assets/logo.png';
import 'theme/Login.css';
import 'theme/Forms.css';
import Loader from 'components/core/cards/dashboard/Loader';

const Errors = ({ errors }) => {
  if (!Object.keys(errors).length > 0) return null;

  return (
    <ul className="errors">
      {Object.keys(errors)
        .filter(e => !!errors[e])
        .map(e => <li key={e}>{errors[e]}</li>)}
    </ul>
  );
};

Errors.propTypes = {
  errors: PropTypes.object
};

class Login extends React.Component {
  state = {
    isLoading: false,
    errors: null
  };

  onLoginSubmit({ username, password }) {
    this.setState({ isLoading: true });
    this.props.auth(username, password).catch(err => {
      this.setState({
        isLoading: false,
        errors: { server: `There was a problem: ${err.message}` }
      });
    });
  }

  render() {
    return (
      <section className="login-section">
        <div className="login-card">
          {this.state.isLoading ? (
            <Loader />
          ) : (
            <React.Fragment>
              <img className="logo" src={logo} alt="Clear logo" />
              <Form onSubmit={values => this.onLoginSubmit(values)}>
                {({ submitForm, errors }) => (
                  <form onSubmit={submitForm}>
                    <Errors errors={{ ...this.props.errors, ...this.state.errors, ...errors }} />
                    <Text field="username" placeholder="Your username" validateOnSubmit required />
                    <Text
                      field="password"
                      type="password"
                      placeholder="*********"
                      validateOnSubmit
                      required
                    />
                    <button type="submit" className="btn">
                      Login
                    </button>
                  </form>
                )}
              </Form>
            </React.Fragment>
          )}
        </div>
      </section>
    );
  }
}

Login.propTypes = {
  errors: PropTypes.object,
  auth: PropTypes.func.isRequired
};

Login.defaultValues = {
  errors: {}
};

export default Login;
