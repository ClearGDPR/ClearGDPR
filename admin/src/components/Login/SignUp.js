import React from 'react';

export default class SignUp extends React.PureComponent {
  state = {
    isLoading: false,
    name: null,
    email: null,
    password: null
  };

  onSignUp(e) {
    e.preventDefault();
    this.setState({ isLoading: true });
    fetch('/management/users/register', {
      method: 'POST',
      body: JSON.stringify({
        name: this.state.name,
        email: this.state.email,
        password: this.state.password
      }),
      headers: {
        'Content-Type': 'application/json'
      }
    })
      .then(res => {
        this.setState({ isLoading: false });
      })
      .catch(err => {
        console.log(err);
      });
  }

  render() {
    const { state } = this;

    return (
      <form onSubmit={e => this.onSignUp(e)}>
        <input type="text" ref="name" value={state.name} placeholder="Full Name" />
        <input type="text" ref="email" value={state.email} placeholder="Your email" />
        <input type="password" ref="password" value={state.password} placeholder="*********" />
        <button type="submit" className={`button ${state.isLoading ? 'is-loading' : ''}`}>
          Sign Up
        </button>
      </form>
    );
  }
}
