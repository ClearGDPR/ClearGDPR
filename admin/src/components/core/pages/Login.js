import React from 'react';
import Particles from 'react-particles-js';
import LoginCard from 'components/core/cards/LoginCard';
import './login.css';

const Login = () => {
  return (
    <section className="login-section">
      <LoginCard />
      <Particles
        style={{
          position: 'absolute',
          width: '100%',
          height: '100%',
          top: 0,
          bottom: 0,
          left: 0,
          right: 0,
          pointerEvents: 'none',
          zIndex: '-100',
          opacity: '1'
        }}
        params={{
          particles: {
            dots: {
              color: '#82efa6'
            },
            line_linked: {
              color: '#82efa6'
            }
          }
        }}
      />
    </section>
  );
};

export default Login;
