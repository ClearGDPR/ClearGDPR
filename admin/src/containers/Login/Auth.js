import cookie from 'cookie';
import config from '../../config';

export default class Auth {
  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);
  }

  login(username, password) {
    fetch(`${config.API_URL}/management/users/login`, {
      method: 'POST',
      body: JSON.stringify({
        email: username,
        password: password
      }),
      headers: {
        'Content-Type': 'application/json'
      }
    })
      .then(res => {
        this.setSession(res);
      })
      .catch(err => {
        return err;
      });
  }

  setSession(authResult) {
    const maxAge = 1 * 24 * 60 * 60; // 1 day long;
    document.cookie = cookie.serialize('auth', authResult, { maxAge });
    document.cookie = cookie.serialize('expires_at', authResult, { maxAge });
  }

  logout() {
    document.cookie = cookie.serialize('auth', '', { maxAge: -1 });
    document.cookie = cookie.serialize('expires_at', '', { maxAge: -1 });
  }

  isAuthenticated() {
    let expiresAt = cookie.parse(document.cookie).expires_at || '';
    return new Date().getTime() < expiresAt;
  }
}
