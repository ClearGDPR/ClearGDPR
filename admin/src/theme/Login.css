/* LOGIN SECTION */

.login-section {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.login-card {
  display: flex;
  flex-direction: column;
  background-color: var(--color-background);
  padding: 2rem;
  box-shadow: 0 10px 20px var(--block-fx-shadow);
  min-width: 285px;
  min-height: 325px;
}

.login-card > .logo {
  width: 220px;
  margin: var(--spacing) 0;
}

.login-card > p {
  text-align: center;
  margin-bottom: 0;
}

.login-card > p > small {
  text-decoration: underline;
  font-size: 0.7rem;
}

.login-card > form {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: var(--spacing) 0;
}

.login-card > form > input {
  font-size: 0.8rem;
  width: 100%;
  margin: 0.2rem 0;
  padding: 0.8rem 0.6rem;
}

.login-card .btn {
  text-align: center;
  text-decoration: none;
  font-size: 0.8rem;
  text-transform: uppercase;
  padding: var(--spacing);
  background-color: var(--color-foreground);
  color: var(--color-background);
}
