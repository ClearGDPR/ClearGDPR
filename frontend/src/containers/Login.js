/* eslint react/prop-types: 0 */
import React, { Component } from 'react';
import config from '../config';

const { API_BASE } = config;

class Login extends Component {
  state = {
    isLoading: false
  };

  onLoginHandler = e => {
    e.preventDefault();
    this.setState({ isLoading: true });

    const { email, password } = this.refs;
    const url = API_BASE + '/api/users/login';

    fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        username: email.value,
        password: password.value
      })
    })
      .then(res => res.json())
      .then(token => {
        this.setState({ isLoading: false });
        localStorage.setItem('cgToken', token.cgToken);
        this.props.history.push('/profile');
      })
      .catch(console.log);
  };

  render() {
    return (
      <section className="section" id="give-consent">
        <div className="container">
          <div className="columns">
            <div className="column is-one-third">
              <h3 className="title"> Sign in to your account </h3>
              <hr />
              <form onSubmit={this.onLoginHandler}>
                <div className="field">
                  <label className="label">Email</label>
                  <div className="control">
                    <input className="input" ref="email" type="email" required />
                  </div>
                </div>
                <div className="field">
                  <label className="label">Your Password</label>
                  <div className="control">
                    <input className="input" ref="password" type="password" />
                  </div>
                </div>

                <button
                  type="submit"
                  className={`button is-primary ${this.state.isLoading ? 'is-loading' : ''}`}
                >
                  Sign in
                </button>
              </form>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default Login;
