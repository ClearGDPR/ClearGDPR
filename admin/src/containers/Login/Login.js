import React from 'react';
import logo from '../../assets/logo.png';
import Login from '../../components/Login/Login';
import session from '../../helpers/Session';
import config from '../../config';
import '../../theme/Login.css';

const handleLogin = (username, password) => {
  fetch(`${config.API_URL}/management/users/login`, {
    method: 'POST',
    body: JSON.stringify({
      email: username,
      password: password
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  })
    .then(res => {
      console.log(res);
      session.set(res);
    })
    .catch(err => {
      return err;
    });
};

const LoginContainer = () => (
  <section className="login-section">
    <div className="login-card">
      <img className="logo" src={logo} alt="Clear logo" />
      <Login auth={handleLogin} />
    </div>
  </section>
);

export default LoginContainer;
