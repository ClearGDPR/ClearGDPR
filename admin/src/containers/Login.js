import React from 'react';
import SignUp from '../components/Login/SignUp';

const LoginContainer = () => <SignUp />;

export default LoginContainer;
