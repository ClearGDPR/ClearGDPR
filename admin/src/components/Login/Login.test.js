import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './Login';

const setup = propOverrides => {
  const props = Object.assign({
    auth: jest.fn()
  }, propOverrides);

  const component = shallow(
    <Login {...props} />
  );

  return { props, component };
};


describe('(Component) Login', () => {
  it('should render correctly', () => {
    const { component } = setup();
    expect(toJson(component)).toMatchSnapshot();
  });
});
