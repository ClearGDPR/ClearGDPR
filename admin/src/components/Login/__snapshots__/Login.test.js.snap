// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Login should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    auth={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-card"
      >
        <img
          alt="Clear logo"
          className="logo"
          src="logo.png"
        />
        <form
          onSubmit={[Function]}
        >
          <input
            placeholder="Your username"
            type="text"
          />
          <input
            placeholder="*********"
            type="password"
          />
          <button
            className="btn"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>,
      "className": "login-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Clear logo"
            className="logo"
            src="logo.png"
          />,
          <form
            onSubmit={[Function]}
          >
            <input
              placeholder="Your username"
              type="text"
            />
            <input
              placeholder="*********"
              type="password"
            />
            <button
              className="btn"
              type="submit"
            >
              Login
            </button>
          </form>,
        ],
        "className": "login-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Clear logo",
            "className": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                placeholder="Your username"
                type="text"
              />,
              <input
                placeholder="*********"
                type="password"
              />,
              <button
                className="btn"
                type="submit"
              >
                Login
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Your username",
                "type": "text",
              },
              "ref": "username",
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "*********",
                "type": "password",
              },
              "ref": "password",
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-card"
        >
          <img
            alt="Clear logo"
            className="logo"
            src="logo.png"
          />
          <form
            onSubmit={[Function]}
          >
            <input
              placeholder="Your username"
              type="text"
            />
            <input
              placeholder="*********"
              type="password"
            />
            <button
              className="btn"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>,
        "className": "login-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Clear logo"
              className="logo"
              src="logo.png"
            />,
            <form
              onSubmit={[Function]}
            >
              <input
                placeholder="Your username"
                type="text"
              />
              <input
                placeholder="*********"
                type="password"
              />
              <button
                className="btn"
                type="submit"
              >
                Login
              </button>
            </form>,
          ],
          "className": "login-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Clear logo",
              "className": "logo",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  placeholder="Your username"
                  type="text"
                />,
                <input
                  placeholder="*********"
                  type="password"
                />,
                <button
                  className="btn"
                  type="submit"
                >
                  Login
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Your username",
                  "type": "text",
                },
                "ref": "username",
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "*********",
                  "type": "password",
                },
                "ref": "password",
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "btn",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
